Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -221,11 +250,46 @@ class SyncNowSetting: WithAccountSetting {
     override func onClick(_ navigationController: UINavigationController?) {
         NotificationCenter.default.post(name: Notification.Name(rawValue: SyncNowSetting.NotificationUserInitiatedSyncManually), object: nil)
         profile.syncManager.syncEverything(why: .syncNow)
+        
+        if AppConstants.MOZ_FXA_PREFERENCE {
+            profile.getAccount()?.updateProfile()
+        }
     }
 }
 
 // Sync setting that shows the current Firefox Account status.
 class AccountStatusSetting: WithAccountSetting {
+    override init(settings: SettingsTableViewController) {
+        super.init(settings: settings)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(AccountStatusSetting.updateAccount(notification:)), name: NotificationFirefoxAccountProfileChanged, object: nil)
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: NotificationFirefoxAccountProfileChanged, object: nil)
+    }
+    
+    func updateAccount(notification: Notification) {
+        DispatchQueue.main.async {
+            self.settings.tableView.reloadData()
+        }
+    }
+    
+    override var image: UIImage? {
+        if !AppConstants.MOZ_FXA_PREFERENCE {
+            return nil
+        }
+        
+        var image: UIImage?
+        if let account = profile.getAccount(), let imageData = account.fxaProfile?.avatarImageData {
```

Comment:
As implied earlier, this would be cleaner if you created the image when you downloaded the data. The reason `image` is optional is because `UIImage(data:)` is failable; you get neater code and no fallibility here if you construct the image when you download the data.